lib/src/merged_tree.rs

```diff
@@ -891,9 +892,59 @@ impl MergedTreeBuilder {
                 }
                 Ok(MergedTreeId::Legacy(tree_builder.write_tree()))
             }
-            MergedTreeId::Merge(_) => {
-                todo!()
+            MergedTreeId::Merge(base_tree_ids) => {
+                match base_tree_ids.into_resolved() {
+                    Ok(base_tree_id) => {
+                        // Optimized case for the common case of writing a single tree
+                        let mut tree_builder =
+                            TreeBuilder::new(self.store.clone(), base_tree_id.clone());
+                        for (path, values) in self.overrides {
+                            tree_builder.set_or_remove(path, values.into_resolved().unwrap());
+                        }
+                        let tree_id = tree_builder.write_tree();
+                        Ok(MergedTreeId::resolved(tree_id))
```

Comment:
How well this optimized path work? I think it's practically the same as `write_merged_trees([base_tree_id])` (though compiler can generate slightly better code for the optimized path.)

Reply:
Good question. I think my main concern was the cloning of each path and value. I clarified that in the comment. It might still not have any measurable effect (this is not used in a write-heavy code path until I've integrated the rebase code with it). I don't mind dropping it until we can profile it, if you prefer.

Reply:
Ok, thanks. I'm not sure if that would have measurable difference, but I'm not against adding simpler path for the common case.

Can you also add a test that covers this code path? I think the added tests only cover legacy and conflicting cases.

Reply:
Thinking more about this, I can't imagine this optimization would have any measurable impact. It just avoids cloning paths and values for modified paths, but most paths are not typically not modified, and we clone the entire trees in several places already. I'll drop this optimization instead.